src/plugins/workspace/server/utils.ts

```diff
@@ -50,3 +54,63 @@ export const getPrincipalsFromRequest = (
 
   throw new Error('UNEXPECTED_AUTHORIZATION_STATUS');
 };
+
+export const updateDashboardAdminStateForRequest = (
+  request: OpenSearchDashboardsRequest,
+  groups: string[],
+  users: string[],
+  configGroups: string[],
+  configUsers: string[]
+) => {
+  if (configGroups.length === 0 && configUsers.length === 0) {
+    updateWorkspaceState(request, {
+      isDashboardAdmin: false,
+    });
+    return;
+  }
+  const groupMatchAny = groups.some((group) => configGroups.includes(group)) || false;
+  const userMatchAny = users.some((user) => configUsers.includes(user)) || false;
+  updateWorkspaceState(request, {
+    isDashboardAdmin: groupMatchAny || userMatchAny,
+  });
+};
+
+export const stringToArray = (adminConfig: string | undefined) => {
+  if (!adminConfig) {
+    return [];
+  }
+  let adminConfigArray;
+  try {
+    adminConfigArray = JSON.parse(adminConfig);
+  } catch (e) {
+    return [];
+  }
+  return adminConfigArray;
+};
+
+export const getApplicationOSDAdminConfig = async (
+  { applicationConfig }: AppPluginSetupDependencies,
+  request: OpenSearchDashboardsRequest
+) => {
+  const applicationConfigClient = applicationConfig.getConfigurationClient(request);
+
+  const [groupsResult, usersResult] = await Promise.all([
+    applicationConfigClient
+      .getEntityConfig('opensearchDashboards.dashboardAdmin.groups')
+      .catch(() => undefined),
+    applicationConfigClient
+      .getEntityConfig('opensearchDashboards.dashboardAdmin.users')
+      .catch(() => undefined),
+  ]);
+
+  return [stringToArray(groupsResult), stringToArray(usersResult)];
+};
+
+export const getOSDAdminConfig = async (globalConfig$: Observable<SharedGlobalConfig>) => {
+  const globalConfig = await globalConfig$.pipe(first()).toPromise();
+  const groupsResult = (globalConfig.opensearchDashboards?.dashboardAdmin?.groups ||
+    []) as string[];
```

Comment:
```suggestion
    []);
```
Is the type assertion `as string[]` necessary here?

Reply:
Yes, `as string[]` can make the return type consistent when calling the method in `src/plugins/workspace/server/plugin.ts`(106 line).